tests/test_component/cpp/ArrayMethods.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+#include "ArrayMethods.g.h"
+
+namespace winrt::test_component::implementation
+{
+    struct ArrayMethods
+    {
+        ArrayMethods() = default;
+
+        static hstring InInt32Array(array_view<int32_t const> value);
+        static hstring InStringArray(array_view<hstring const> value);
+        static hstring InObjectArray(array_view<winrt::Windows::Foundation::IInspectable const> value);
+        static hstring InStringableArray(array_view<winrt::Windows::Foundation::IStringable const> value);
+        static hstring InStructArray(array_view<winrt::test_component::BlittableStruct const> value);
+        static hstring InNonBlittableStructArray(array_view<winrt::test_component::NonBlittableStruct const> value);
+        static hstring InEnumArray(array_view<winrt::test_component::Signed const> value);
+
+    private:
+
+        static hstring InEnum(winrt::test_component::Signed const& value);
+
+        template<typename T>
+        static void simulate_rpc_behavior(array_view<T> const& value)
+        {
+            // RPC requires array pointers to be non-null.
+            if (value.begin() == nullptr)
```

Comment:
Curious, where in WinRT is this important?

Reply:
this was an artifact of using the same test component that C++/WinRT used. i imagine from the comment that it's when arrays are marshaled cross-proc via RPC